api/routes.go

```diff
@@ -71,5 +71,5 @@ func RegisterRoutes() {
 	shared.AddRoute("/api/screenshot/{screenshot:.*}", "api-screenshot", shared.WrapPermissiveCORS(apiScreenshotRedirectHandler))
 
 	// API endpoint for searching Metadata for the products.
-	shared.AddRoute("/api/metadata", "api-metadata", shared.WrapPermissiveCORS(apiMetadataHandler))
+	shared.AddRoute("/api/metadata", "api-metadata", shared.WrapRestrictiveCORS(apiMetadataHandler, []string{"https://developer.mozilla.org"}, []string{"GET", "POST", "PATCH", "OPTIONS"}))
```

Comment:
Thanks! I split the endpoint according to the methods it can receive. 

Regarding whether to list OPTIONS as an accepted header - I do see numerous examples [ search Access-Control-Allow-Methods on this page](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) on MDN which list OPTIONS, and gorilla does not seem to set OPTIONS either.